packages/wormhole-sdk/src/bridge.ts

```diff
@@ -42,40 +49,71 @@ export class WormholeBridge {
   settings: AllBridgeSettings
 
   constructor({ srcDomain, dstDomain, srcDomainProvider, dstDomainProvider, settings }: WormholeBridgeOpts) {
-    this.srcDomain = srcDomain
+    this.srcDomain = getLikelyDomainId(srcDomain)
     this.dstDomain = dstDomain || getDefaultDstDomain(srcDomain)
     this.srcDomainProvider = srcDomainProvider || new ethers.providers.JsonRpcProvider(DEFAULT_RPC_URLS[this.srcDomain])
     this.dstDomainProvider = dstDomainProvider || new ethers.providers.JsonRpcProvider(DEFAULT_RPC_URLS[this.dstDomain])
 
     this.settings = { useFakeArbitrumOutbox: false, ...settings }
   }
 
+  private async _optionallySendTx(
+    shouldSendTx: boolean,
+    contract: Contract,
+    method: string,
+    data: any[],
+    overrides?: Overrides,
+  ): Promise<Call> {
+    return {
+      tx: shouldSendTx ? await contract[method](...data, { ...overrides }) : undefined,
+      to: contract.address,
+      data: contract.interface.encodeFunctionData(method, data),
+    }
+  }
+
   public async initWormhole(
-    sender: Signer,
     receiverAddress: string,
     amount: BigNumberish,
     operatorAddress?: string,
+    sender?: Signer,
     overrides?: Overrides,
-  ): Promise<ContractTransaction> {
+  ): Promise<Call> {
+    const shouldSendTx = Boolean(sender)
+    sender ||= Wallet.createRandom().connect(this.srcDomainProvider)
+
     const dstDomainBytes32 = bytes32(this.dstDomain)
     const sender_ = sender.provider ? sender : sender.connect(this.srcDomainProvider)
 
     const sdk = getSdk(this.srcDomain, sender_)
     const l2Bridge = sdk.WormholeOutboundGateway!
 
     if (operatorAddress) {
-      return l2Bridge['initiateWormhole(bytes32,address,uint128,address)'](
-        dstDomainBytes32,
-        receiverAddress,
-        amount,
-        operatorAddress,
-        { ...overrides },
+      return await this._optionallySendTx(
+        shouldSendTx,
+        l2Bridge,
+        'initiateWormhole(bytes32,address,uint128,address)',
+        [dstDomainBytes32, receiverAddress, amount, operatorAddress],
+        overrides,
       )
     }
 
-    return l2Bridge['initiateWormhole(bytes32,address,uint128)'](dstDomainBytes32, receiverAddress, amount, {
-      ...overrides,
-    })
+    return await this._optionallySendTx(
+      shouldSendTx,
+      l2Bridge,
+      'initiateWormhole(bytes32,address,uint128)',
+      [dstDomainBytes32, receiverAddress, amount],
+      overrides,
+    )
+  }
+
+  public async initRelayedWormhole(
+    receiverAddress: string,
+    amount: BigNumberish,
+    sender?: Signer,
+    overrides?: Overrides,
+  ): Promise<Call> {
+    const dstDomainSdk = getSdk(this.dstDomain, Wallet.createRandom().connect(this.dstDomainProvider))
```

Comment:
a much better pattern to create a read only SDK would be to use `this.dstDomainProvider as any`. (This should be improved on `eth-sdk` side...)

Reply:
Agreed and changed